task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -717,7 +717,11 @@ def _push_xcom_if_needed(result: Any, ti: RuntimeTaskInstance):
     _xcom_push(ti, "return_value", result, mapped_length=mapped_length)
 
 
-def finalize(log: Logger): ...
+def finalize(ti: RuntimeTaskInstance, log: Logger):
+    for oe in ti.task.operator_extra_link_dict.values():
+        link, xcom_key = oe.get_link(operator=ti.task, ti_key=ti.id), oe.xcom_key  # type: ignore[arg-type]
+        log.debug("Setting xcom for operator extra link", link=link, xcom_key=xcom_key)
+        _xcom_push(ti, key=xcom_key, value=link)
```

Comment:
Question on previous behaviuor? Where links available for failed tasks? I guess they were (in which case this is the right place to do it.)

Reply:
Precisely, thats the behaviour. And also the reason why I kept this in finalize.

There's a use case, a link to "how to write a dag effectively" if a dag times out due to being poorly written :)